runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkStreamingPortablePipelineTranslator.java

```diff
@@ -170,7 +172,12 @@ public FlinkPipelineOptions getPipelineOptions() {
 
     @Override
     public JobExecutionResult execute(String jobName) throws Exception {
-      return getExecutionEnvironment().execute(jobName);
+      try {
+        return getExecutionEnvironment().execute(jobName);
+      } catch (OptimizerPlanEnvironment.ProgramAbortException ex) {
```

Comment:
this needs to move to `FlinkPipelineRunner`

Reply:
Yes, that would be the better place.